File: frontend/src/types/document.ts
export interface DocumentResponse {
  id: string;
  claim_id: string;
  document_type: string;
  file_path: string;
  quality_score: number;
  is_duplicate: boolean;
  created_at: string; // ISO date string
  file_name?: string;
  file_size?: number; // in bytes
  mime_type?: string;
}

File: app/schemas/document.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import Optional, List

# Option to use strict Enum, but user said string is OK
class DocumentUploadResponse(BaseModel):
    id: UUID
    claim_id: UUID
    document_type: str
    file_path: str
    quality_score: int
    is_duplicate: bool
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class DocumentListResponse(BaseModel):
    documents: List[DocumentUploadResponse]
